src/librustdoc/lib.rs

```diff
@@ -741,6 +749,32 @@ fn run_renderer<'tcx, T: formats::FormatRenderer<'tcx>>(
     }
 }
 
+fn generate_book(render_options: &mut config::RenderOptions) -> Result<(), String> {
+    let Some(config::PathOrURL::Path(ref mut book_dir)) = render_options.book_location else {
+        return Ok(());
+    };
+    if !book_dir.is_dir() {
+        return Err(format!(
+            "`{}` is not a folder, expected a folder or a URL for `--book-location` argument",
+            book_dir.display(),
+        ));
+    }
+    let mut book = match mdbook::MDBook::load(&book_dir) {
+        Ok(book) => book,
+        Err(error) => return Err(format!("failed to load book: {error:?}")),
+    };
+    let output_dir = render_options.output.join("doc-book");
+    *book_dir = output_dir.join("index.html");
```

Comment:
I don't like that this mutates the `render_options.book_location` to record the new path of the book. I think it'd better to have two separate fields, one for the source location of the book and one for the target location (None if the source was a URL).

Reply:
Not sure to agree but not something I think is worth debating about so let's go. However: I'm gonna use `PathOrUrl` because otherwise you'd need to use two types to get the information, which is suboptimal imo.